imports/ui/unit-explorer/filtered-units.jsx

```diff
@@ -5,24 +5,40 @@ import UnitTypeIcon from './unit-type-icon'
 
 import { resetMenuItemDivStyle } from '../general.mui-styles'
 
-export default class UnitsByRoles extends Component {
+export default class FilteredUnits extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      ready: false
+    }
+  }
+
+  componentDidUpdate (prevProps) {
+    const { searchResult } = this.props
+    if ((prevProps.searchResult && prevProps.searchResult.length) !== (searchResult && searchResult.length)) {
+      this.setState({ready: true})
+    }
+  }
+
   render () {
-    const { unitsByRoles, showAddBtn, handleUnitClicked, handleAddCaseClicked, administrate } = this.props
+    const { unitsByRoles, searchResult, showAddBtn, handleUnitClicked, handleAddCaseClicked, administrate } = this.props
+    const { ready } = this.state
+    const filteredUnits = unitsByRoles || searchResult
     return (
-      <div>
-        {(unitsByRoles.length !== 0) && (
+      <div className='flex-grow flex flex-column overflow-auto'>
+        {(filteredUnits.length !== 0 && searchResult === undefined) && (
           <div className='bb b--black-10 bg-very-light-gray f6 fw5 mid-gray pv2 pl2'>
             { administrate ? ('Units I Administrate') : ('Units Involved In') }
           </div>)
         }
-        {
-          unitsByRoles.map(({ id, name, description, metaData }) => (
+        { ((ready && searchResult) || unitsByRoles) &&
+          filteredUnits.map(({ id, name, description, metaData }) => (
             <MenuItem key={id} innerDivStyle={resetMenuItemDivStyle} onClick={() => handleUnitClicked(id)} >
               <div className='mt2 ph2 bb b--very-light-gray br1 w-100 flex items-center pa2'>
                 <UnitTypeIcon metaData={metaData} />
                 <div className='ml3 mv1 semi-dark-gray lh-copy flex-grow overflow-hidden'>
-                  <div className='ti1 ellipsis'>{metaData.displayName || name}</div>
-                  <div className='ti1 ellipsis silver'>{ metaData.moreInfo || description}&nbsp;</div>
+                  <div className='ti1 ellipsis'>{name}</div>
```

Comment:
Why was the metaData usage removed? These attributes should be populated by the UnitsMetaData mongo collection whenever possible